be/src/exec/vectorized/hdfs_scanner.h

```diff
@@ -272,6 +272,8 @@ class HdfsScanner {
     std::vector<ExprContext*> _min_max_conjunct_ctxs;
     std::unique_ptr<RandomAccessFile> _raw_file;
     std::unique_ptr<RandomAccessFile> _file;
+    // by default it's no compression.
+    CompressionTypePB _compression_type = CompressionTypePB::NO_COMPRESSION;
```

Comment:
If this _compression_type only be updated and used for text scanner, it may not be appropriate to put it in the base class, which may be  confused with compression type of other formats.

Reply:
`_compression_type` is used only by hdfs text scanner, however random access file is opened in base clas `open_random_access_file`. This base function will use `compression_type`.

```
Status HdfsScanner::open_random_access_file() {
    CHECK(_file == nullptr) << "File has already been opened";
    ASSIGN_OR_RETURN(_raw_file, _scanner_params.fs->new_random_access_file(_scanner_params.path))
    auto stream = std::make_shared<CountedSeekableInputStream>(_raw_file->stream(), &_stats);
    if (_compression_type == CompressionTypePB::NO_COMPRESSION) {
        _file = std::make_unique<RandomAccessFile>(stream, _raw_file->filename());
    } else {
        using DecompressorPtr = std::shared_ptr<StreamCompression>;
        std::unique_ptr<StreamCompression> dec;
        RETURN_IF_ERROR(StreamCompression::create_decompressor(_compression_type, &dec));
        auto compressed_input_stream =
                std::make_shared<io::CompressedInputStream>(stream, DecompressorPtr(dec.release()));
        auto compressed_seekable_input_stream =
                std::make_shared<io::CompressedSeekableInputStream>(compressed_input_stream);
        _file = std::make_unique<RandomAccessFile>(compressed_seekable_input_stream, _raw_file->filename());
    }
    return Status::OK();
}
```

Right now `_compression_type` is used only by hdfs text scanner, but I think it's also ok to put in base class and more scanners will benefit from that.

Reply:
> `_compression_type` is used only by hdfs text scanner, however random access file is opened in base clas `open_random_access_file`. This base function will use `compression_type`.
> 
> ```
> Status HdfsScanner::open_random_access_file() {
>     CHECK(_file == nullptr) << "File has already been opened";
>     ASSIGN_OR_RETURN(_raw_file, _scanner_params.fs->new_random_access_file(_scanner_params.path))
>     auto stream = std::make_shared<CountedSeekableInputStream>(_raw_file->stream(), &_stats);
>     if (_compression_type == CompressionTypePB::NO_COMPRESSION) {
>         _file = std::make_unique<RandomAccessFile>(stream, _raw_file->filename());
>     } else {
>         using DecompressorPtr = std::shared_ptr<StreamCompression>;
>         std::unique_ptr<StreamCompression> dec;
>         RETURN_IF_ERROR(StreamCompression::create_decompressor(_compression_type, &dec));
>         auto compressed_input_stream =
>                 std::make_shared<io::CompressedInputStream>(stream, DecompressorPtr(dec.release()));
>         auto compressed_seekable_input_stream =
>                 std::make_shared<io::CompressedSeekableInputStream>(compressed_input_stream);
>         _file = std::make_unique<RandomAccessFile>(compressed_seekable_input_stream, _raw_file->filename());
>     }
>     return Status::OK();
> }
> ```
> 
> Right now `_compression_type` is used only by hdfs text scanner, but I think it's also ok to put in base class and more scanners will benefit from that.

OK